department-of-veterans-affairs/caseflow
# APPEALS-41109 ACD Distribution Statistics (#20954)

Resolves [APPEALS-41109](https://jira.devops.va.gov/browse/APPEALS-41109)

# Description
Adds `distribution_stats` database table and stores most of a distribution's statistics in that table instead of the `statistics` column on the `distributions` table.
* Note: a distribution's statistics field still contains a JSON object with the `batch_size` field, which is required for calculating the priority target in the weekly PushPriorityAppealsToJudges job

## Acceptance Criteria
- [x] New DB table exists
- [x] Distribution statistics are stored in the `distribution_stats` table
- [x] The PushPriorityAppealsToJudges job runs and correctly calculates the Priority Target using previous distribution data

## Testing Plan
1. Run `make migrate` locally
2. Verify `distribution_stats` table is created
3. In a Rails console, run the Push Priority job: `PushPriorityAppealsToJudgesJob.new.perform_now`
4. Verify in the database that the distribution record's statistics columns contain `batch_size` and `message` fields
5. Verify in the database that each distribution has an associated `distribution_stats` record

# Backend
## Database Changes
*Only for Schema Changes*


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
